test/k8sT/istio.go

```diff
@@ -0,0 +1,325 @@
+		Expect(err).To(BeNil(), "Cilium cannot be installed")
+
+		ExpectCiliumReady(kubectl)
+		ExpectKubeDNSReady(kubectl)
+
+		By("Creating the istio-system namespace")
+		res := kubectl.NamespaceCreate(istioSystemNamespace)
+		res.ExpectSuccess("unable to create namespace %q", istioSystemNamespace)
+
+		By("Creating the Istio resources")
+		res = kubectl.Apply(istioYAMLPath)
+		res.ExpectSuccess("unable to create Istio resources")
+
+		// Ignore one-time jobs and Prometheus. All other pods in the
+		// namespaces have an "istio" label.
+		By("Waiting for Istio pods to be ready")
+		err = kubectl.WaitforPods(istioSystemNamespace, "-l istio", 300)
+		Expect(err).To(BeNil(),
+			"Istio pods are not ready after timeout in namespace %q", istioSystemNamespace)
+
+		for _, name := range istioServiceNames {
+			By("Waiting for Istio service %q to be ready", name)
+			err = kubectl.WaitForServiceEndpoints(
+				istioSystemNamespace, "", name, helpers.HelperTimeout)
+			Expect(err).Should(BeNil(), "Service %q is not ready after timeout", name)
+		}
+
+		for _, name := range istioServiceNames {
+			By("Waiting for DNS to resolve Istio service %q", name)
+			err = kubectl.WaitForKubeDNSEntry(name, istioSystemNamespace)
+			Expect(err).To(BeNil(), "DNS entry is not ready after timeout")
+		}
+	})
+
+	AfterAll(func() {
+		By("Deleting the Istio resources")
+		_ = kubectl.Delete(istioYAMLPath)
+
+		By("Deleting the istio-system namespace")
+		_ = kubectl.NamespaceDelete(istioSystemNamespace)
+	})
+
+	JustBeforeEach(func() {
+		var err error
+		err, microscopeCancel = kubectl.MicroscopeStart()
+		Expect(err).To(BeNil(), "Microscope cannot be started")
+
+		uptimeCancel, err = kubectl.BackgroundReport("uptime")
+		Expect(err).To(BeNil(), "Cannot start background report process")
+	})
+
+	JustAfterEach(func() {
+		Expect(microscopeCancel()).To(BeNil(), "Cannot stop microscope")
+		uptimeCancel()
+
+		kubectl.ValidateNoErrorsOnLogs(CurrentGinkgoTestDescription().Duration)
+	})
+
+	AfterFailed(func() {
+		kubectl.CiliumReport(helpers.KubeSystemNamespace,
+			"cilium endpoint list",
+			"cilium bpf proxy list")
+	})
+
+	// This is a subset of Services's "Bookinfo Demo" test suite, with the pods
+	// injected with Istio sidecar proxies and Istio mTLS enabled.
+	Context("Istio Bookinfo Demo", func() {
+
+		var (
+			resourceYAMLPaths []string
+			policyPaths       []string
+		)
+
+		BeforeEach(func() {
+			// Those YAML files are the bookinfo-v1.yaml and bookinfo-v2.yaml
+			// manifests injected with Istio sidecars using those commands:
+			// istioctl kube-inject -f bookinfo-v1.yaml > bookinfo-v1-istio.yaml
+			// istioctl kube-inject -f bookinfo-v2.yaml > bookinfo-v2-istio.yaml
+			bookinfoV1YAML := helpers.ManifestGet("bookinfo-v1-istio.yaml")
+			bookinfoV2YAML := helpers.ManifestGet("bookinfo-v2-istio.yaml")
+
+			// This YAML is copied from examples/kubernetes-istio.
+			initPolicyPath := helpers.ManifestGet("istio-sidecar-init-policy.yaml")
```

Comment:
Not-blocking: we have had issues before where when we copy files from other locations in the repo, they get out of sync with the CI, which results in us not testing what is in our documentation that we push out for users. This can be done as follow-up / cleanup.

Reply:
How can I fix that? What's the proper pattern?